arrow/src/test/java/org/apache/iceberg/arrow/vectorized/ArrowReaderTest.java

```diff
@@ -909,7 +907,7 @@ private DataFile writeParquetFile(Table table, List<GenericRecord> records) thro
     }
 
     PartitionKey partitionKey = new PartitionKey(table.spec(), table.schema());
-    partitionKey.partition(new LocalDateTimeToLongMicros(records.get(0)));
+    partitionKey.partition(records.get(0));
```

Comment:
This was previously correct. This needs to translate between the representations.

Reply:
so that means https://github.com/apache/iceberg/issues/6510 is not a bug but users have to do the conversion themselves?